core/modules/link/link.module

```diff
@@ -335,6 +349,11 @@ function link_field_widget_form(&$form, &$form_state, $field, $instance, $langco
     '#type' => $instance['widget']['type'],
     '#default_value' => isset($items[$delta]) ? $items[$delta] : '',
   );
+  if ($instance['settings']['type'] != LINK_BOTH) {
+    $element['#description'] .= ' ' . t('Only %type URLs are allowed.', array(
```

Comment:
Should this always be there - even if the site builder decided to not add any help text?

Anyway, it probably should only show up, if validation in general is turned on.

Reply:
> Should this always be there - even if the site builder decided to not add any help text?

Not sure... Happy to hear what others have to say. Discuss in https://github.com/backdrop/backdrop-issues/issues/5678 ?

Reply:
Ah, nevermind. Just saw @quicksketch's comment below. Will remove this line then.

Reply:
@BWPanda I don't think that @quicksketch read @indigoxela's comment right. @indigoxela suggested that we should be adding this help text _always_ - not to remove it altogether. Not having this does not provide any hints whatsoever for the content editors, which is not good UX:
![image](https://user-images.githubusercontent.com/2423362/217204586-27a60f27-c85a-467c-87ec-525d8f05924c.png)
The above should have some help text like the following (intentionally not using the more technical term "URL" here):
- internal only: `Only internal links are allowed. Start typing to see a list of suggestions.`
- external only: `Only external links are allowed.`
- both: `Enter an external link, or start typing to see a list of suggestions of internal links.`